### ISSUE TITLE
Add level-of-detail / length control

### ISSUE THREAD

**Issue by waldyrious** [waldyrious/primerpedia#19]
There could be a set of radio buttons, a dropdown list or even a slider (UI up for discussion) to pick between different content lengths:
- [clause](https://en.wikipedia.org/wiki/Sentence_%28linguistics%29#Clauses)
  - structured
    - [Wikidata description](https://www.wikidata.org/wiki/Help:Description#Length)
    - [short descriptions](https://en.wikipedia.org/wiki/Wikipedia:Short_description)
  - unstructured
    - [disambiguation pages](https://en.wikipedia.org/wiki/MOS:DABENTRIES)
    - [hatnotes](https://en.wikipedia.org/wiki/Wikipedia:Hatnote)
    - ["See also" sections](https://en.wikipedia.org/wiki/MOS:SEEALSO)
- sentence ([WP:LEADSENTENCE](https://en.wikipedia.org/wiki/WP:LEADSENTENCE))
- paragraph ([WP:LEADPARAGRAPH](https://en.wikipedia.org/wiki/WP:LEADPARAGRAPH))
- section ([WP:LEADSECTION](https://en.wikipedia.org/wiki/WP:LEADSECTION))

**Comment by waldyrious** [comment]
Changing the control should update the currently loaded article, not just future queries.

**Comment by Jan-Ka** [comment]
I tried to find a solution for this.

The control is not an issue but I have problems with understand how to obtain the information from wikipedia/mediaWiki. Any pointers?

**Comment by waldyrious** [comment]
Yeah, that's why I removed the `hacktoberfest` label, as this probably needs a bit of research work. I would say that a first implementation could be done by fetching the contents as is currently done, and simply trimming out excess contents (just the first sentence, just the first paragraph, and the entire first section). This could be done using regex find/replace.

Eventually, we will want to also fetch the Wikidata description. but I guess that could be left to a separate issue, to keep things simple.

**Comment by waldyrious** [comment]
By the way: Back in 2012 I made a [proposal for related functionality](https://en.wikipedia.org/wiki/MediaWiki_talk:Gadget-popups.js/Archive_1#New_feature_proposal:_lead_only) in Wikipedia's [Navigation popups](https://en.wikipedia.org/wiki/Wikipedia:Tools/Navigation_popups) gadget. Some of the code there may be relevant for this. Here's the [full current source](https://en.wikipedia.org/wiki/MediaWiki:Gadget-popups.js) of the gadget.

**Comment by waldyrious** [comment]
Funny thing, I just learned about Ted Nelson's (of Project Xanadu fame) [StretchText](https://en.wikipedia.org/wiki/StretchText) proposal, which is a proposal for precisely this kind of [progressive disclosure](https://en.wikipedia.org/wiki/Progressive_disclosure) in prose (though it's a bit more fleshed-out in terms of how the UI should work, and more ambitious in the seamlessness of the intended UX).
